src/rpc/RPCHelpers.cpp

```diff
@@ -1478,4 +1478,210 @@ specifiesCurrentOrClosedLedger(boost::json::object const& request)
     return false;
 }
 
+std::variant<ripple::uint256, Status>
+getNFTID(boost::json::object const& request)
+{
+    if (!request.contains(JS(nft_id)))
+        return Status{Error::rpcINVALID_PARAMS, "missingTokenid"};
+
+    if (!request.at(JS(nft_id)).is_string())
+        return Status{Error::rpcINVALID_PARAMS, "tokenidNotString"};
+
+    ripple::uint256 tokenid;
+    if (!tokenid.parseHex(request.at(JS(nft_id)).as_string().c_str()))
+        return Status{Error::rpcINVALID_PARAMS, "malformedCursor"};
+
+    return tokenid;
+}
+
+// TODO - this function is long and shouldn't be responsible for as much as it
+// is. Split it out into some helper functions.
+std::variant<Status, boost::json::object>
+traverseTransactions(
+    Context const& context,
+    std::function<Backend::TransactionsAndCursor(
+        std::uint32_t const,
+        bool const,
+        std::optional<Backend::TransactionsCursor> const&)> getter)
+{
+    auto request = context.params;
+    boost::json::object response = {};
+
+    bool const binary = getBool(request, JS(binary), false);
+    bool const forward = getBool(request, JS(forward), false);
+
+    std::optional<Backend::TransactionsCursor> cursor;
+
+    if (request.contains(JS(marker)))
+    {
+        auto const& obj = request.at(JS(marker)).as_object();
```

Comment:
nice call - it is not (this was existing code but going to fix it)

Reply:
so, i've fixed this. but interestingly i just checked what rippled does in this case and it hard errors just like we did - IE it's just an "internalError". That's obviously not right but @cjcobb23 to what extent should we be _exactly_ like rippled here?

Reply:
my $0.02 is that this is an example of where clio should be different from rippled. the shape of the happy path response is the same, but we are improving errors. rippled should later have a project to incorporate these improvements there.

Reply:
We should avoid `internalError` at all costs, and I don't think it's documented that a non-object marker results in `internalError`. We actually are striving to adhere to the documented API exactly, as opposed to rippled itself. So we have some freedom here, since I don't think documentation says anything about non-object markers. I think we should improve the errors and return something like `markerNotObject`